server/src/internalClusterTest/java/org/opensearch/indices/memory/breaker/ConcurrentSearchRethrowExceptionIT.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.memory.breaker;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.search.ShardSearchFailure;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.core.common.breaker.CircuitBreaker;
+import org.opensearch.core.common.breaker.CircuitBreakingException;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.script.MockScriptPlugin;
+import org.opensearch.script.Script;
+import org.opensearch.script.ScriptType;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.opensearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE;
+import static org.opensearch.index.query.QueryBuilders.scriptQuery;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE)
+public class ConcurrentSearchRethrowExceptionIT extends OpenSearchIntegTestCase {
+
+    @Override
+    protected Settings featureFlagSettings() {
+        return Settings.builder().put(super.featureFlagSettings()).put(FeatureFlags.CONCURRENT_SEGMENT_SEARCH, "true").build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singleton(ScriptedExceptionPlugin.class);
+    }
+
+    public void testShardFailureException() {
+        client().prepareIndex("test").setId("1").setSource("field", "value").setRefreshPolicy(IMMEDIATE).get();
+
+        SearchResponse searchResponse = client().prepareSearch("test")
+            .setQuery(
```

Comment:
The test case does the job but is not representative of what is really happening, @jed326 do you think we could simulate the issue by triggering CB during search operation?

Reply:
@jed326 thats not true, this test case goes to the `rethrowCauseIfPossible` method and then tests it. So, its not true that we are not testing the code.

@reta yes the issue could be simulated by testing the same set of tests with concurrent search enabled and number of slices=1. I was thinking in this case, I can add all the tests in the `CircuitBreakerServiceIT.java` with concurrent search enabled and the slice count value = 1. Thoughts?

Reply:
>  I was thinking in this case, I can add another class with all the tests in the CircuitBreakerServiceIT.java with concurrent search enabled and the slice count value = 1.

All for it, thank you, as far as we get realistic flow tested